source/core/NumericString.cpp

```diff
@@ -1351,6 +1351,31 @@ void DoubleScanString(StaticTypeAndData* argument, TypeRef argumentType, TempSta
 }
 
 //----------------------------------------------------------------------------------------------------
+Boolean EnumScanString(SubString* in, StaticTypeAndData* argument, TypeRef argumentType, char formatChar, char* beginPointer, char** endPointer)
+{
+    Boolean found = false;
+    if (formatChar == 's') {
+        for (IntIndex i = 0; i < argumentType->GetEnumItemCount(); i++) {
+            SubString itemName = argumentType->GetEnumItemName(i)->MakeSubStringAlias();
+            if (in->ComparePrefix(itemName.Begin(), itemName.Length()))
```

Comment:
This will incorrectly take the first match instead of the longest match if the enum strings themselves have any common prefixes.  For example, scanning a string with an Enum16 ("red" "redorange" "redyellow") would match "red" first and so never match the other two.